### PR TITLE
Update NuGet tasks to new auth style

### PR DESCRIPTION
**Pull request by jmyersmsft**
Reimplemented in Node to avoid rewriting them twice when we have to support them xplat

### CODE REVIEW

**Comment by jmyersmsft** [comment]
@emanuelquintero @IvyMH Could you take a look

@bryanmacfarlane now that Matt Cooper has left the Packaging team, I'm not sure who if anyone on our team can complete this PR, or even how to find out

File: Tasks/Common/nuget-task-common/LocationHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+                var spsUri = getUriForServiceDefinition(sps);
+                tl.debug(util.format("found SPS at %s", spsUri));
+
+                return getConnectionData(spsUri, accessToken)
+                    .then(spsConnectionData => {
+                        tl.debug("successfully loaded SPS location data");
+                        var areaService = findServiceByIdentifier(spsConnectionData, areaId);
+                        if (!areaService) {
+                            var error = new Error("Could not locate nuget service in the root location service");
```
Comment (emanuelquintero) [inline Tasks/Common/nuget-task-common/LocationHelpers.ts]: I don't see anything in the code that's nuget specific. So maybe change the string?

File: Tasks/Common/nuget-task-common/LocationHelpers.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+export function getConnectionDataForArea(serviceUri: string, areaName: string, areaId: string, accessToken: string): Q.Promise<locationApi.ConnectionData> {
+    return getConnectionData(serviceUri, accessToken)
+        .then(connectionData => {
+            tl.debug("successfully loaded origin service location data");
+            if (hasServicesOfType(connectionData, areaName)) {
+                tl.debug(util.format("found %s routes on %s", areaName, serviceUri));
+                return Q(connectionData);
+            }
+            else {
+                tl.debug(util.format("did not find %s routes directly on %s, trying SPS", areaName, serviceUri));
```
Comment (emanuelquintero) [inline Tasks/Common/nuget-task-common/LocationHelpers.ts]: Should the strings be localized?

File: Tasks/NugetPublisher/nugetpublisher.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+});
+
+var connectedServiceName = tl.getInput('connectedServiceName');
+var internalFeedUri = tl.getInput('feedName');
+var nuGetAdditionalArgs = tl.getInput('nuGetAdditionalArgs');
+var verbosity = tl.getInput('verbosity');
+var preCredProviderNuGet = tl.getBoolInput('preCredProviderNuGet');
+
+var nuGetFeedType = tl.getInput('nuGetFeedType') || "external";
+// normalize the restore mode for display purposes, and ensure it's a known one
```
Comment (emanuelquintero) [inline Tasks/NugetPublisher/nugetpublisher.ts]: Comment should be removed

**Comment by emanuelquintero** [comment]
@jmyersmsft left some comments
:shipit:

**Pull request merged**